### PR TITLE
feat: spq_tally()

### LINKED ISSUE #94: n() auxiliary for spq_summarise()
The function spq_summarise() does not allow the use of n() (for COUNT). The following query does not work for instance:
```
library(glitter)
tib=spq_init() %>%
spq_add("?film wdt:P31 wd:Q11424",.label="?film") %>%  
spq_add("?film wdt:P840 ?loc",.label="?loc") %>% 
spq_summarise(n_films=n()) %>% 
spq_perform() 
```

### PR DESCRIPTION
**Pull request by maelle**
Fix #94 (we need a separate issue for `n_distinct()`)

I decided to try and implement a tally function because I worried about making `spq_summarize()` too complicated. What do you think?

Do we also need an `spq_count()` function? Do we still need to amend `spq_summarize()`?

``` r
library("glitter")
spq_init() %>%
spq_add("?film wdt:P31 wd:Q11424") %>%
spq_mutate(narrative_location = wdt::P840(film), .label = "?narrative_location") %>%
spq_tally(name = "n_films") %>%
spq_perform()
#> # A tibble: 1 × 1
#>   n_films
#>     <dbl>
#> 1   50940

spq_init() %>%
spq_add("?film wdt:P31 wd:Q11424",.label="?film") %>%
spq_mutate(narrative_location = wdt::P840(film), .label = "?narrative_location") %>%
spq_group_by(narrative_locationLabel) %>%
spq_tally(sort = TRUE, name = "n_films") %>%
spq_perform()
#> # A tibble: 2,701 × 2
#>    n_films narrative_locationLabel
#>      <dbl> <chr>                  
#>  1    3118 Indonesia              
#>  2    2540 New York City          
#>  3    2220 London                 
#>  4    1851 Paris                  
#>  5    1488 Los Angeles            
#>  6    1427 England                
#>  7     964 Rome                   
#>  8     837 California             
#>  9     792 France                 
#> 10     705 Hong Kong              
#> # ℹ 2,691 more rows
```

<sup>Created on 2023-05-26 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### CODE REVIEW

**Comment by lvaudor** [comment]
Well... I had never seen tally() to this day, and I rarely use count() : I'd rather use group_by() => summarise (n=n()). In short(!) I guess I have trouble -to some extent- with shortcuts (a.k.a. lack of mental flexibility?). 

What do you mean when you say that spq_summarize() could become too complicated? complicated to develop or to use?

**Comment by lvaudor** [comment]
Are you referring to the amendment regarding n() and n_distinct()? If so, I'd like to keep the amendment regarding n(). On the other hand n_distinct() is not frequently used in my experience -??- so maybe we could pass on this one or at least consider it a low priority. I like your spq_tally (now that I know what tally is, and now that you have implemented it, why not keep it ;-) ) and in my experience an spq_count() could be even more useful (although to be honest both might be especially useful to people other than myself).

**Comment by maelle** [comment]
reg n_distinct see #115 , the feature request isn't clear to me on that point, and we can tackle it in another PR.

**Comment by maelle** [comment]
two special cases to add to `spq_summarize()`: `n()` without arguments on its own, `n()` without arguments but accompanied by other things.

**Comment by maelle** [comment]
another TODO for myself: with the third example in the tests, I'm pretty sure loc should be in SELECT. related to #117

**Comment by maelle** [comment]
@lvaudor things should eventually work, but next week :smile_cat:

File: R/spq_summarise.R [hunk 1]
```diff
@@ -3,16 +3,17 @@
...
 #' spq_add("?item wdt:P1082 ?folkm_ngd") %>%
 #' spq_add("?area wdt:P31 wd:Q1907114", .label = "?area") %>%
 #' spq_add("?area wdt:P527 ?item") %>%
 #' spq_group_by(area, areaLabel)  %>%
 #' spq_summarise(total_folkm = sum(folkm_ngd))
-spq_summarise = function(.query, ...){
+spq_summarise = function(.query, ...) {
 
   variables = purrr::map_chr(rlang::enquos(...), spq_treat_argument)
+  are_we_tallying <- (variables == "COUNT(*)")
```
Comment (maelle) [inline R/spq_summarise.R:16]: argh, should probably be an %in% as we might be creating other variables at the same time

File: R/spq_group_by.R [hunk 2]
```diff
@@ -12,13 +12,38 @@
...
   varformula = purrr::map_df(.query$select, get_varformula)
-  .query$select = varformula %>%
-    dplyr::filter(.data$name %in% vars) %>%
-    dplyr::pull(.data$full)
-  .query$group_by = varformula %>%
-    dplyr::filter(.data$name %in% {{ vars }}) %>%
-    dplyr::pull(.data$formula)
+
+  vars_selected = vars[vars %in% varformula[["name"]]]
+  vars_formulaed <- dplyr::filter(
```
Comment (maelle) [inline R/spq_group_by.R:20]: I think that file might be refactored later, when we better keep track of variables used in formulas.

**Comment by lvaudor** [comment]
Yes! That's great! Well done

**Pull request merged**